dev/tests/static/framework/Magento/CodeMessDetector/Rule/Design/CookieAndSessionMisuse.php

```diff
@@ -105,7 +108,8 @@ private function isControllerPlugin(\ReflectionClass $class): bool
         foreach ($class->getMethods(\ReflectionMethod::IS_PUBLIC) as $method) {
             if (preg_match('/^(after|around|before).+/i', $method->getName())) {
                 try {
-                    $argument = $method->getParameters()[0]->getClass();
+                    $parameter = $method->getParameters()[0];
+                    $argument = $this->getParameterClass($parameter);
```

Comment:
Let's avoid one time use variables if possible
```suggestion
                    $argument = $this->getParameterClass($method->getParameters()[0]);
```